oshi-core/src/main/java/oshi/software/os/mac/MacOperatingSystem.java

```diff
@@ -393,4 +396,92 @@ public NetworkParams getNetworkParams() {
         return new MacNetworkParams();
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public OSService[] getServices() {
+        //Sort by PID
+        OSProcess[] process =  getChildProcesses(1, 0, OperatingSystem.ProcessSort.PID);
+        //Get Directories
+        File ua = new File("~/Library/LaunchAgents");
+        File ga = new File("/Library/LaunchAgents");      
+        File gd = new File("/Library/LaunchDaemons");      
+        File sa = new File("/System/Library/LaunchAgents");      
+        File sd = new File("/System/Library/LaunchDaemons");
+
+        ArrayList<File> files = new ArrayList<File>();
+
+        try {
+            File[] uaFiles = ua.listFiles((dir, name) -> name.toLowerCase().endsWith(".plist"));
+            File[] gaFiles = ga.listFiles((dir, name) -> name.toLowerCase().endsWith(".plist"));
+            File[] gdFiles = gd.listFiles((dir, name) -> name.toLowerCase().endsWith(".plist"));
+            File[] saFiles = sa.listFiles((dir, name) -> name.toLowerCase().endsWith(".plist"));
+            File[] sdFiles = sd.listFiles((dir, name) -> name.toLowerCase().endsWith(".plist"));
+
+            for(File file : uaFiles) files.add(file);
+            for(File file : gaFiles) files.add(file);
+            for(File file : gdFiles) files.add(file);
+            for(File file : saFiles) files.add(file);
+            for(File file : sdFiles) files.add(file);
+
+            OSService[] svcArray = new OSService[files.size()];
+            for (int i = 0; i < svcArray.length; i++) {
+                for (int j = 0; j < process.length; j++) {
+                    svcArray[i].setName(files.get(i).getName()); 
```

Comment:
You haven't initialized `svcArray[i]` so this throws a NPE here.  Also I suspect it should be outside the iteration over j.